File: package/main/src/Random/randomFloat.ts
/**
 * Returns a random float in the half-open interval `[min, max)`.
 * @param {number} min - Inclusive lower bound
 * @param {number} max - Exclusive upper bound
 * @returns {number} Random float
 * @example
 * randomFloat(0, 1); // 0.000... up to but not including 1
 */
export const randomFloat = (min: number, max: number): number =>
  Math.random() * (max - min) + min;

File: package/umt_python/src/random/random_float.py
import random as random_module


def random_float(min_value: float, max_value: float) -> float:
    """
    Returns a random float in the half-open interval [min_value, max_value).

    Args:
        min_value: Inclusive lower bound.
        max_value: Exclusive upper bound.

    Returns:
        A random float.

    Example:
        >>> random_float(7, 7)
        7.0
        >>> 0 <= random_float(0, 1) < 1
        True
    """
    return random_module.random() * (max_value - min_value) + min_value
